utils/kubernetes/deployment.go

```diff
@@ -52,12 +52,12 @@ func IsDeploymentProgressing(deployment *appsv1.Deployment) bool {
 	return false
 }
 
-// GetDeploymentUnavailabilityReason returns a string explaining why the given deployment is unavailable. If empty, there's no replica failure.
+// GetDeploymentUnavailabilityMessage returns a string explaining why the given deployment is unavailable. If empty, there's no replica failure.
 // Note that the Deployment might be available, but a second replica failed to scale. Always check IsDeploymentAvailable.
-func GetDeploymentUnavailabilityReason(deployment *appsv1.Deployment) string {
+func GetDeploymentUnavailabilityMessage(deployment *appsv1.Deployment) string {
 	for _, condition := range deployment.Status.Conditions {
-		if condition.Type == appsv1.DeploymentReplicaFailure &&
-			condition.Status == v1.ConditionTrue {
+		if condition.Type == appsv1.DeploymentAvailable &&
```

Comment:
Shouldn't be the deployment in status ```DeploymentUnavailableReason```

Reply:
It might! I'll take a look at the docs again.

Reply:
Actually, no. But I improved this function to look for two conditions in the underlying deployment. This way we can provide a more precise error message/reason to a failure.